src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previous;
+        private Node<T> next;
+
+        private Node(T value, Node<T> previous, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
+    private int size;
+    private Node<T> headNode;
+    private Node<T> tailNode;
+
+    private Node<T> getNodeFromIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("wrong index"
+                    + "\n"
+                    + "Mast be from 0 to " + (size() - 1));
+        }
+        Node<T> searchNode = headNode;
+
+        if (index < (size >> 1)) {
+            for (int i = 0; i < index; i++) {
+                searchNode = searchNode.next;
+            }
+        } else {
+            searchNode = tailNode;
+            for (int i = size() - 1; i > index; i--) {
+                searchNode = searchNode.previous;
+            }
+        }
+        return searchNode;
+
+    }
+
+    private void addToEmpty(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        headNode = newNode;
+        tailNode = newNode;
+        size++;
+    }
+
+    private void addFirst(T value) {
+        if (size == 0) {
+            addToEmpty(value);
+        } else {
+            Node<T> newNode = new Node<>(value, null, headNode);
+            headNode.previous = newNode;
+            headNode = newNode;
+            size++;
+        }
+
+    }
+
     @Override
-    public void add(T value) {
 
+    public void add(T value) {
+        if (size == 0) {
+            addToEmpty(value);
+        } else {
+            Node<T> newNode = new Node<>(value, tailNode, null);
+            tailNode.next = newNode;
+            tailNode = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("wrong index"
+                    + "\n" + "Mast be from 0 to "
+                    + (size() - 1));
+        }
+        if (size() == 0) {
+            addToEmpty(value);
+        } else if (index == 0) {
+            addFirst(value);
+        } else if (index == size()) {
```

Comment:
```suggestion
        } else if (index == size() || size() == 0) {
```
and then remove 
if (size() == 0) {
            addToEmpty(value);
        }

Reply:
total agree